src/Core/Domain/Order/ValueObject/OrderDetailId.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Order\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderException;
+
+/**
+ * Order identity
+ */
+class OrderDetailId
+{
+    /**
+     * @var int
+     */
+    private $orderDetailId;
+
+    /**
+     * @param int $orderDetailId
+     *
+     * @throws OrderException
+     */
+    public function __construct(int $orderDetailId)
+    {
+        $this->assertIntegerIsGreaterThanZero($orderDetailId);
+
+        $this->orderDetailId = $orderDetailId;
+    }
+
+    /**
+     * @return int
+     */
+    public function getValue(): int
+    {
+        return $this->orderDetailId;
+    }
+
+    /**
+     * @param int $orderId
+     */
+    private function assertIntegerIsGreaterThanZero(int $orderId): void
+    {
+        if (0 > $orderId) {
+            throw new OrderException(sprintf('Order detail id %s is invalid. Order detail id must be number that is greater than zero.', var_export($orderId, true)));
```

Comment:
var export is not needed. It was used some time ago when we didn't have strict types. Now you know it is of type int, so no point of that anymore.
```suggestion
            throw new OrderException(sprintf('Order detail id %d is invalid. Order detail id must be number that is greater than zero.', $orderId));
```
Not blocking though